packages/database/test/exp/integration.test.ts

```diff
@@ -114,109 +118,134 @@ describe('Database@exp Tests', () => {
 
   // Tests to make sure onValue's data does not get mutated after calling get
   it('calls onValue only once after get request with a non-default query', async () => {
-    const db = getDatabase(defaultApp);
-    const { ref: testRef } = getUniqueRef(db);
+    const { readerRef } = getRWRefs(getDatabase(defaultApp));
     const queries = [
-      query(testRef, limitToFirst(1)),
-      query(testRef, startAt('child1')),
-      query(testRef, startAt('child2')),
-      query(testRef, limitToFirst(2))
+      query(readerRef, limitToFirst(1)),
+      query(readerRef, startAt('child1')),
+      query(readerRef, startAt('child2')),
+      query(readerRef, limitToFirst(2))
     ];
     await Promise.all(
       queries.map(async q => {
         const initial = [{ name: 'child1' }, { name: 'child2' }];
         const ec = EventAccumulatorFactory.waitsForExactCount(1);
-
-        await set(testRef, initial);
-        const unsubscribe = onValue(testRef, snapshot => {
-          ec.addEvent(snapshot.val());
+        const writerPath = getFreshRepo(readerRef._path);
+        await set(writerPath, initial);
+        const unsubscribe = onValue(readerRef, snapshot => {
+          ec.addEvent(snapshot);
         });
         await get(q);
         await waitFor(2000);
         const [snap] = await ec.promise;
-        expect(snap).to.deep.equal(initial);
+        expect(snap.val()).to.deep.equal(initial);
         unsubscribe();
       })
     );
   });
 
+  it('calls onValue() listener when get() is called on a parent node', async () => {
+    // Test that when get() is pending on a parent node, and then onValue is called on a child node, that after the get() comes back, the onValue() listener fires.
+    const db = getDatabase(defaultApp);
+    const { readerRef, writerRef } = getRWRefs(db);
+    await set(writerRef, {
+      foo1: {
+        a: 1
+      },
+      foo2: {
+        b: 1
+      }
+    });
+    let resolved = false;
+    get(readerRef).then(() => (resolved = true));
+    const childPath = child(readerRef, 'foo1');
+    expect(resolved).to.be.false;
```

Comment:
Lets add a comment explaining why it's difficult to test the behavior we're trying to test and then convert this into a smoke test (just run this in a loop a few times).

Reply:
But we should get rid of this assertion because it can't always pass).

Reply:
SGTM